src/Mimi.jl

```diff
@@ -192,11 +192,18 @@ function setindex{T}(m::Model, name::Symbol, values::Vector{T})
     nothing
 end
 
-function addcomponent(m::Model, t, name::Symbol;before=nothing,after=nothing)
+"""
+Add a component to a model.
+"""
+function addcomponent(m::Model, t, name::Symbol=:nothing; before=nothing,after=nothing)
```

Comment:
Can you make the name argument ``name::Union{Symbol,Void}=nothing``? I'm not a fan of sentinel values. My suggestion will make the function type-instable, but I think we can ignore performance implications here.